console.log("try catch");

try {
    // let a = 'Adi';
    // // a = undefined;
    // if (a != undefined) {
    //     throw new Error('This is not undefined');
    // } else {
    //     console.log('This is undefined');

    // }
    asjkdcnakscmasc
} catch (error) {
    console.log('U have generated the -> ' + error + " error");
    console.log(error.name);
    console.log(error.message);
    

} finally{
    // runs irrespective of try or catch
    console.log('Finally');
    
}